tests/app/celery/test_letters_pdf_tasks.py

```diff
@@ -646,6 +646,54 @@ def test_process_sanitised_letter_with_valid_letter(
     assert file_contents == 'sanitised_pdf_content'
 
 
+@mock_s3
+@pytest.mark.parametrize('address, expected_postage, expected_international',
+                         [('Lady Lou, 123 Main Street, SW1 1AA', 'second', False),
+                          ('Lady Lou, 123 Main Street, France', 'europe', True),
+                          ('Lady Lou, 123 Main Street, New Zealand', 'rest-of-world', True),
+                          ])
+def test_process_sanitised_letter_sets_postage_international(
+    sample_letter_notification,
+    expected_postage,
+    expected_international,
+    address
+):
+    filename = 'NOTIFY.{}'.format(sample_letter_notification.reference)
+
+    scan_bucket_name = current_app.config['LETTERS_SCAN_BUCKET_NAME']
+    template_preview_bucket_name = current_app.config['LETTER_SANITISE_BUCKET_NAME']
+    destination_bucket_name = current_app.config['LETTERS_PDF_BUCKET_NAME']
+    conn = boto3.resource('s3', region_name='eu-west-1')
+    conn.create_bucket(Bucket=scan_bucket_name)
+    conn.create_bucket(Bucket=template_preview_bucket_name)
+    conn.create_bucket(Bucket=destination_bucket_name)
+
+    s3 = boto3.client('s3', region_name='eu-west-1')
+    s3.put_object(Bucket=scan_bucket_name, Key=filename, Body=b'original_pdf_content')
+    s3.put_object(Bucket=template_preview_bucket_name, Key=filename, Body=b'sanitised_pdf_content')
```

Comment:
Shouldn’t this be mocked? Feels a bit weird to be connecting to an external service that could be unavailable from an integration test.

Reply:
That's what @mock_s3 does

Reply:
Oh, that’s fancy